### PR TITLE
legacy: WebSession removal

### PR DESCRIPTION
**Pull request by hachreak**
* Adds a new reset password form.

* Enhances the password changing form and registration form with new
  validators.

* Removes not used ChangeUserEmailSettingsForm form.

* Adds check email and check nickname function inside model and rewrites
  form validators to use them.

* Rewrites legacy code for lost password and reset password pages.

* Adds Jinja templates for reset password page and email.

* INCOMPATIBLE Removes legacy WebSession module. (addresses #3233)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio/modules/accounts/forms.py [hunk 1]
```diff
@@ -113,6 +107,20 @@ def _validator(form, field):
     return _validator
 
 
+def current_user_password_validator(form, field):
+    """Validate password field if is the password of the user."""
+    id = current_user.get_id()
+    if not id:
+        raise validators.ValidationError(
+            _("Nobody is currently logged-in"))
```
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: add `.` at the end

File: invenio/modules/accounts/forms.py [hunk 2]
```diff
@@ -207,7 +182,28 @@ class LostPasswordForm(InvenioBaseForm):
...
+    """Form to change password."""
+
+    password = PasswordField(
+        _("New password"),
+        description=_("The password phrase may contain punctuation, "
+                      "spaces, etc."),
+        validators=[
+            DataRequired(),
+            min_length_password_validator,
+            EqualTo('password2', message=_('Passwords must match'))]
```
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: `Both passwords must match.`

File: invenio/modules/accounts/models.py [hunk 3]
```diff
@@ -84,6 +85,21 @@ def __str__(self):
     PROFILE_FIELDS = ['nickname', 'email', 'family_name', 'given_names']
     """List of fields that can be updated with update_profile."""
 
+    @staticmethod
+    def check_nickname(nickname):
+        """Check if it's a valid nickname."""
+        re_invalid_nickname = re.compile(""".*[,'@]+.*""")
```
Comment (jirikuncar) [inline invenio/modules/accounts/models.py]: can you check the CERN oauthclient?

Comment (hachreak) [inline invenio/modules/accounts/models.py]: done.

**Comment by hachreak** [comment]
ping @tiborsimko - upgrade recipe added! :shipit:

**Comment by hachreak** [comment]
ping @jirikuncar

File: invenio/modules/accounts/forms.py [hunk 4]
```diff
@@ -113,6 +107,20 @@ def _validator(form, field):
     return _validator
 
 
+def current_user_password_validator(form, field):
+    """Validate password field if is the password of the user."""
+    id = current_user.get_id()
```
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: please call it `uid` (or `id_user`) so you don't redefine `id` function.

File: invenio/modules/accounts/forms.py [hunk 5]
```diff
@@ -139,59 +147,26 @@ def validate_login_method(self, field):
...
     """Profile form."""
 
     nickname = StringField(
         _("Username"),
-        validators=[DataRequired(), current_user_validator('nickname'),
-                    nickname_validator]
+        validators=[
+            DataRequired(),
+            current_user_validator('nickname'),
+            nickname_validator]
```
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: WDYT about following style: `s/]/,\n        ]/`?

File: invenio/modules/accounts/forms.py (hunk 2 above)
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: @hachreak ping!

File: invenio/modules/accounts/forms.py [hunk 6]
```diff
@@ -250,11 +256,14 @@ class RegisterForm(Form):
         description=_(
             "The password phrase may contain punctuation, spaces, etc."
         ),
-        validators=[password_validator],
+        validators=[
+            DataRequired(),
+            EqualTo('password2', message=_('Passwords must match')),
```
Comment (jirikuncar) [inline invenio/modules/accounts/forms.py]: see also https://github.com/inveniosoftware/invenio/pull/3326/files#r33690355

File: invenio/modules/accounts/views/accounts.py [hunk 7]
```diff
@@ -30,24 +30,33 @@
...
 from invenio.legacy import webuser
-from invenio.modules.access.mailcookie import mail_cookie_check_mail_activation
+from invenio.modules.access.errors import \
+    InvenioWebAccessMailCookieDeletedError, InvenioWebAccessMailCookieError
+from invenio.modules.access.mailcookie import \
+    mail_cookie_check_mail_activation, mail_cookie_check_pw_reset, \
+    mail_cookie_delete_cookie
+from invenio.modules.accounts.forms import LoginForm, LostPasswordForm, \
+    RegisterForm, ResetPasswordForm
+from invenio.modules.accounts.models import User
```
Comment (jirikuncar) [inline invenio/modules/accounts/views/accounts.py]: please use relative imports: `from .models import User`

**Pull request merged**